### ISSUE TITLE
Cannot resolve version ranges

### ISSUE THREAD

**Issue by dibben-jsol** [conan-io/conan#1352]
I am getting the following error:
`Error: The version in '~=1' could not be resolved.`
when running conan install on a package.

The relevant requires line in the conanfile.py is
 `requires = FileHandler/[~=1]@builduser/snapshot`

If I replace this line by 
  `requires = FileHandler/1.0.0@builduser/snapshot`
Then it works without error.

If the 1.0.0 version of the library is installed locally it also works with the revision range. However, if it is not installed locally, we use Artifactory (5.3.1) as a conan server then it does not work.

Confusingly, some packages work, others do not. It actually seems if the package contains "Handler" in the package name the it does not work, otherwise it does.

Looking at the conan trace file the call to Artifactory is as follows
`http://xx.xx.xx.xx:8081/artifactory/api/conan/libs-snapshot/v1/conans/search?q=FileHandler/*@builduser/snapshot`
When run in a browser this returns
```
{
  "results" : [ ]
}
```
However, running
`http://xx.xx.xx.xx:8081/artifactory/api/conan/libs-snapshot/v1/conans/search?q=FileHandler/1.0.0@builduser/snapshot`
returns
```
{
  "results" : [ "FileHandler/1.0.0@builduser/snapshot" ]
}
```
The server only contains a single version for this library.
For a different library, the "*"  search works
```
http://xx.xx.xx.xx:8081/artifactory/api/conan/libs-snapshot/v1/conans/search?q=filesep/*@builduser/snapshot

{
  "results" : [ "filesep/1.0.0@builduser/snapshot" ]
}
```

Is this an Artifactory issue? I tried several different library modules. If the form is XXXHandler then the "*" search does not work. So far all the other libraries it seems to work. Cannot imagine how this is relevant but I suppose it might help.

My assumption on the correct behaviour is that the [~=1] range gets mapped to the latest 1.x.x library. This will then be installed locally. So the revision range, in this case, should map to 1.0.0

**Comment by memsharded** [comment]
Thanks for the detailed report. It could be a minor mismatch in the Artifactory API, we will investigate asap.

**Comment by memsharded** [comment]
Confirmed. It is working fine with conan_server, but failing in Artifactory (5.2.1 the one I have tried).
Starting to work on fixing the issue, but Artifactory might take a bit longer to release, so please try to use the exact version until this is fixed. Thanks very much!

**Comment by memsharded** [comment]
More info: You were right, it is something related with the package name in the search method. Basically, the "and" (case insensitive) is detected as a logical AND clause in the search query. You should be fine avoiding names with those letters when using version ranges. So the recommendation to use the exact version in the meantime is still valid.

A fix has already been submitted to Artifactory, I will update with releases here.
Thanks again for your report, very useful.

**Comment by dibben-jsol** [comment]
Thanks!. I look forward to getting the fixed version of Artifactory.

**Comment by memsharded** [comment]
Hi @dibben-jsol 

This issue was solved and released in 5.5: https://www.jfrog.com/jira/browse/RTFACT-14338. Did you updated and tried it? It is working? Please tell so we could close this issue. Thanks!

**Comment by dibben-jsol** [comment]
I noticed that it was marked as fixed, so I have been waiting for the 5.5 release.
According to the release notes on the JFrog site, and the bintray download page, 5.4.6 is still the latest version (as of 2017/09/14).  So I have not been able to get 5.5 and therefore not tried it yet.

**Comment by memsharded** [comment]
You are right, sorry, my mistake. Thanks!

**Comment by dibben-jsol** [comment]
We updated to Artifactory 5.5 and it seems to work now.  So it looks like this issue can be closed.
Thanks!

**Comment by memsharded** [comment]
Then something has changed :)
5.5 has not included this fix, it will be in 5.6 (very soon). The issue was related to packages containing the letters "and" in the name, and using version-ranges. Maybe some of them were changed.

**Comment by markgalpin** [comment]
As far as we know this issue is resolved.  If there is a problem with a current version of Artifactory or Bintray please let us know and re-open.
